ui/src/app/store/utils/slice.ts

```diff
@@ -198,23 +149,23 @@ export const generateSlice = <
         // No state changes need to be handled for this action.
       },
     },
-    fetchStart: (state: GenericState<I, E>) => {
+    fetchStart: (state: S) => {
       state.loading = true;
     },
-    fetchError: (state: GenericState<I, E>, action: PayloadAction<E>) => {
+    fetchError: (state: S, action: PayloadAction<S["errors"]>) => {
       state.errors = action.payload;
       if (setErrors) {
         state = setErrors(state, action, "fetch");
       }
       state.loading = false;
     },
-    fetchSuccess: (state: GenericState<I, E>, action: PayloadAction<I[]>) => {
+    fetchSuccess: (state: S, action: PayloadAction<S["items"]>) => {
       state.loading = false;
       state.loaded = true;
       state.items = action.payload;
     },
     create: {
-      prepare: (params: TSFixMe) => ({
+      prepare: (params: Partial<S["items"][0]>) => ({
```

Comment:
Is this type right? When creating something we usually only need to supply a subset of the model's fields - maybe even just a name. Plus in the case of creating machines you can actually pass through things that aren't machine fields, like "don't commission automatically". I think we might need a separate type for the create method if we're going to include it in this util

Reply:
I did that mostly as a convenience, I think we have two choices, either have an extra (optional) generic to supply the type of the create and update params or we define these for each model (like I did here: https://github.com/canonical-web-and-design/maas-ui/pull/2694/files#diff-8d6b0d5a1c7fd502004350766887c23e6a2a8cddcaa8a0c4c3d6052f65c4af7eR241).

What do you reckon?

Reply:
I think defining it for each model makes the most sense. I feel like adding another generic is the wrong abstraction but I can't really explain why - maybe because this PR already does a lot to remove some!

Reply:
OK, are you happy if I unlink https://github.com/canonical-web-and-design/maas-ui/issues/2131 and then follow up by adding the create/update types?

Reply:
Fine by me!